x-pack/legacy/plugins/apm/public/components/app/ServiceOverview/index.tsx

```diff
@@ -70,6 +71,9 @@ export function ServiceOverview() {
     }
   }, [data.hasLegacyData]);
 
+  useTrackPageview({ app: 'apm', path: 'services_overview' });
+  useTrackPageview({ app: 'apm', path: 'services_overview', delay: 15000 });
```

Comment:
Am I correct in understanding that `delay` is there to track people who stayed on the page for longer than 15s?
Say I want to know how long time users are spending on a certain page, and I don't really know what to expect I'd have to track a plethora of durations:

```js
useTrackPageview({ app: 'apm', path: 'services_overview' });
useTrackPageview({ app: 'apm', path: 'services_overview', delay: 5000 });
useTrackPageview({ app: 'apm', path: 'services_overview', delay: 10000 });
useTrackPageview({ app: 'apm', path: 'services_overview', delay: 15000 });
// ...
useTrackPageview({ app: 'apm', path: 'services_overview', delay: 60000 });
```

What about getting rid of `delay` and instead starting a `setInterval` that tracks every 5 seconds. So if the user is on the service overview for 12 seconds the following:
```js
useTrackPageview({ app: 'apm', path: 'services_overview' });
```

would send up:
```
visit_duration_0ms__ services_overview
visit_duration_5000ms__ services_overview
visit_duration_10000ms__ services_overview
```

This should not hurt performance if the interval is 5 seconds, and requests are batched in case users click around wildly (I think batching is necessary regardless).

Reply:
Interesting thoughts! In this case I think it's better to be explicit about the delays you want to track to avoid flooding the telemetry dataset with tons of data points -- basically I'm interested in tracking the difference between "visited page at all" vs "stayed long enough that they _probably_ were there using the app". It's a rough calculation, but you could certainly do something like 10s/20s/30s/1m and even wrap that up in a single hook that wraps `useTrackPageview` if you wanted to simplify.

Could also add a `repeatInterval` prop that would signify wanting to repeat the metric on an interval if you want that behavior, as well. 

Batching: I think would be best handled at the `trackUiMetric` function level perhaps? Do you have thoughts/ideas on what you want to see there?

Reply:
I concur that batching should be implemented in the ui metric service. Ideally it would expose something like a counter to avoid accumulating too many "pings" and flush the counter value to the server regularly and on page unload.

Reply:
@sqren do you have a use-case for wanting to track many varied durations right now or do you think we could start with the explicit path and add repetition later if we discover we want/need it?

Reply:
Yes, I'm good with how it is now. 
I am mostly concerned about the immediate request and whether that will cause lag to other data fetching that happens initially. This could be solved by batching. But I'm okay with not having that in v1